queue/queue.go

```diff
@@ -35,11 +38,31 @@ type MetricsSource interface {
 }
 
 func New() (*Queue, error) {
-	return &Queue{}, nil
+	eventQueue := memqueue.NewQueue(logp.L(), memqueue.Settings{
+		Events: 1024,
+		// The event count and timeout for queue flushes is hard-coded to a placeholder
+		// for now, since that's what the existing Beats API understands. The plan is
+		// for these parameters to instead be controlled by the output as it reads
+		// the queue.
+		FlushMinEvents: 256,
+		FlushTimeout:   5 * time.Millisecond,
+	})
+	producer := eventQueue.Producer(
+		beatsqueue.ProducerConfig{})
+	if producer == nil {
+		fmt.Printf("\033[94mwhy is the queue producer nil\033[0m\n")
```

Comment:
Should not we return an error here?

Reply:
This function can't actually return a nil (it's guaranteed to succeed if the queue object itself is valid), this was just leftover debugging tracking down an uninitialized pointer :-)